packages/@react-spectrum/toast/src/toastContainer.css

```diff
@@ -100,33 +88,57 @@
   flex-direction: inherit;
   align-items: inherit;
   list-style-type: none;
-  margin-block-start: 0;
-  margin-block-end: 0;
-  margin-inline-start: 0;
-  margin-inline-end: 0;
+  margin: 0;
   padding-inline-start: 0;
 }
 
 .spectrum-ToastContainer-listitem {
   display: inline-block;
+  view-transition-class: toast;
 }
 
 @keyframes slide-in {
   from {
-    transform: var(--slide-from);
+    translate: var(--slideX) var(--slideY);
+    opacity: 0;
   }
+}
 
+@keyframes slide-out {
   to {
-    transform: var(--slide-to);
+    translate: var(--slideX) var(--slideY);
+    opacity: 0;
   }
 }
 
-@keyframes fade-out {
-  from {
-    opacity: 1;
-  }
+::view-transition-group(.toast) {
+  animation-duration: 300ms;
+}
 
-  to {
-    opacity: 0;
-  }
+::view-transition-group(.toast.bottom) {
+  --slideX: 0;
+  --slideY: calc(100% + 12px);
+}
+
+::view-transition-group(.toast.top) {
+  --slideX: 0;
+  --slideY: calc(-100% - 12px);
+}
+
+::view-transition-group(.toast.start) {
+  --slideX: calc(-100% - 12px);
+  --slideY: 0;
+}
+
+::view-transition-group(.toast.end) {
+  --slideX: calc(100% + 12px);
+  --slideY: 0;
+}
+
+::view-transition-new(.toast):only-child {
+  animation-name: slide-in;
+}
+
+::view-transition-old(.toast):only-child {
+  animation-name: slide-out;
```

Comment:
Animations look great! Only thing I'd change is that when you close a toast other than the most recent one in the top center/bottom center positions, the toast appears to slide down in front of another toast. Maybe in that case it should only fade out instead of sliding? The sliding works for the "end" placements because it slides out to the right rather than over the top of another toast, and the sliding also works for the very last toast closest to the top/bottom, but the others feel a bit weird. We could potentially special case those with another `view-transition-class`?